src/screens/CityList/CityList.tsx

```diff
@@ -0,0 +1,69 @@
+import type { FC } from "react"
+import { FlatList, StyleSheet, Text } from "react-native"
+import { useNavigation } from "@react-navigation/native"
+import type { StaticScreenProps } from "@react-navigation/native"
+import { useCities } from "../../services/restaurant/hook"
+import { Box, Press, Typography } from "../../components"
+
+type Props = StaticScreenProps<{
+  state: string
+}>
+
+const CityList: FC<Props> = ({ route }) => {
+  const { state } = route.params
+  const navigation = useNavigation()
+  const { data: cities, error, isPending } = useCities(state || "")
+
+  if (error) {
+    return (
+      <Box style={styles.container}>
+        <Text>Error: {error.message}</Text>
```

Comment:
These errors would be more clear if they were something like “Error loading cities: [message]”